src/ert/config/forward_model_step.py

```diff
@@ -46,69 +141,88 @@ class ForwardModelStep:
-            for handle in ["STDOUT", "STDERR", "STDIN"]:
-                if content_dict.get(handle) == "null":
-                    content_dict[handle] = None
-
-            stdout_file = content_dict.get("STDOUT", f"{name}.stdout")
-            stderr_file = content_dict.get("STDERR", f"{name}.stderr")
-
-            return cls(
-                name=name,
-                executable=content_dict.get("EXECUTABLE"),
-                stdin_file=content_dict.get("STDIN"),
-                stdout_file=stdout_file,
-                stderr_file=stderr_file,
-                start_file=content_dict.get("START_FILE"),
-                target_file=content_dict.get("TARGET_FILE"),
-                error_file=content_dict.get("ERROR_FILE"),
-                max_running_minutes=content_dict.get("MAX_RUNNING_MINUTES"),
-                min_arg=content_dict.get("MIN_ARG"),
-                max_arg=content_dict.get("MAX_ARG"),
-                arglist=arglist,
-                arg_types=arg_types_list,
-                environment=environment,
-                required_keywords=required_keywords,
-                exec_env=exec_env,
-                default_mapping=default_mapping,
-                help_text=content_dict.get("HELP_TEXT", ""),
-            )
-        except IOError as err:
-            raise ConfigValidationError.with_context(str(err), config_file) from err
+    def validate_pre_experiment(self, fm_step_json: ForwardModelStepJSON) -> None:
+        """
+        Raise errors pertaining to the environment not being
+        as the forward model step requires it to be. For example
+        a missing FLOW version.
+        """
+
+    def validate_pre_realization_run(
+        self, fm_step_json: ForwardModelStepJSON
+    ) -> ForwardModelStepJSON:
+        """
+        Used to validate/modify the job JSON to be run by the forward model step runner.
+        It will be called for every joblist.json created.
+        """
+        return fm_step_json
+
+    def __post_init__(self) -> None:
+        # We unescape backslash here to keep backwards compatability ie. If
+        # the arglist contains a '\n' we interpret it as a newline.
+        self.arglist = [
+            s.encode("utf-8", "backslashreplace").decode("unicode_escape")
+            for s in self.arglist
+        ]
+
+        self.environment.update(ForwardModelStep.default_env)
+
+        if self.stdout_file is None:
+            self.stdout_file = f"{self.name}.stdout"
+        elif self.stdout_file == "null":
+            self.stdout_file = None
+
+        if self.stderr_file is None:
+            self.stderr_file = f"{self.name}.stderr"
+        elif self.stderr_file == "null":
+            self.stderr_file = None
+
+        if self.stdin_file == "null":
+            self.stdin_file = None
+
+
+class ForwardModelStepPlugin(ForwardModelStep):
+    def __init__(
+        self,
+        name: str,
+        command: List[str],
+        options: Optional[ForwardModelStepOptions] = None,
```

Comment:
`options` here seems like a typical use case for https://typing.readthedocs.io/en/latest/spec/callables.html#unpack-kwargs . I suggest we put in `**kwargs: Unpack[ForwardModelStepOptions]` instead. If a plugin wants to use type the call to their `ForwardModelStepPlugin` we also need to expose `ForwardModelStepOptions` through `ert/__init__.py`.

Reply:
Did not know about `Unpack`, so nice, updated semeio plugins accordingly https://github.com/equinor/semeio/pull/613/commits/055f28ca0aafc7cf8e86678675cc29d3f9e1ece5